### PR TITLE
fix(cloud_identity): google_cloud_identity_group_membership import/update

### PR DESCRIPTION
**Pull request by tmshn**
Fixed bugs in google_cloud_identity_group_membership:

- 36ef27a3d6e1c4de22ac29c91b55b0e9b88036ac: Previously, `terraform import` does not set `group` attribute, which is marked as "ForceNew", so it always show recreation plan after import. Fixed this by filling it on `post_import`
- fcc008fc55210b00caaa86de2f41bf717c4a6bbb: Previously, update functions is not properly implemented, so it always returns 404. Fix this by implementing update function to call `modifyMembershipRoles` method. Fix https://github.com/hashicorp/terraform-provider-google/issues/7616

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
    - 👉 I can't run the exact acctests because I don't have a permission to create groups in my organization. However, I ran them with hardcoding an existing group ID and confirmed they passed.
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
cloud_identity: fix google_cloud_identity_group_membership import/update
```

### CODE REVIEW

File: mmv1/products/cloudidentity/terraform.yaml [hunk 1]
```diff
@@ -77,9 +77,14 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         default_from_api: true
       preferredMemberKey: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
+      roles: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+        set_hash_func: cloudIdentityGroupMembershipRoleHash
```
Comment (rileykarson) [inline mmv1/products/cloudidentity/terraform.yaml:82]: Why do we need a custom hash func here? The default one should do the same.

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb [hunk 2]
```diff
@@ -11,6 +10,165 @@ import (
...
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudIdentityGroupMembershipDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudIdentityGroupMembership_update1(context),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:29]: Can you add an import step after each config step? They're a good way to ensure that each step is stable.

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb [hunk 3]
```diff
@@ -11,6 +10,165 @@ import (
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudIdentityGroupMembership_import(context),
+			},
+			{
+				ResourceName:            "google_cloud_identity_group_membership.basic",
+				ImportState:             true,
+				ImportStateVerify:       true,
+			},
+			{
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:129]: This config step is redundant I think, it should be done implicitly both by the config step & import step before this.

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb [hunk 4]
```diff
@@ -11,6 +10,165 @@ import (
...
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: testAccCheckCloudIdentityGroupMembershipDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudIdentityGroupMembership_import(context),
+			},
+			{
+				ResourceName:            "google_cloud_identity_group_membership.basic",
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:125]: Are we able to remove the `ImportStateVerifyIgnore` on L196 (originally L38) too now?

File: mmv1/products/cloudidentity/terraform.yaml (hunk 1 above)
Comment (tmshn) [reply mmv1/products/cloudidentity/terraform.yaml:82]: I was expecting a new field might be added in the future, and hashing should work correctly after that. The API constraint is:

> Must not contain duplicate MembershipRoles with the same name.

Actually beta version of API has an extra field `expiryDetail`.

https://cloud.google.com/identity/docs/reference/rest/v1beta1/groups.memberships#MembershipRole

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb (hunk 3 above)
Comment (tmshn) [reply mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:129]: Actually I don't know what exactly `ImportStateVerify` does. Is it same as "checking there's no plan diff"?

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb (hunk 2 above)
Comment (tmshn) [reply mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:29]: ✅ 30bc07ed3a7f8ac2e7f6ae0e436d3ece5f0ef25d

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb (hunk 4 above)
Comment (tmshn) [reply mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb:140]: ✅ 30bc07ed3a7f8ac2e7f6ae0e436d3ece5f0ef25d

File: mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb (hunk 3 above)
Comment (tmshn) [reply mmv1/third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb]: ✅ 669c31729867ca3610ba4b3c80687c678fe43236

File: mmv1/products/cloudidentity/terraform.yaml (hunk 1 above)
Comment (rileykarson) [reply mmv1/products/cloudidentity/terraform.yaml:82]: Ah- this is one of provider development's gotchas. The hash function doesn't do _quite_ what most users expect it to. It doesn't identify unique elements in the set- instead, the hash is used to perform the _entire diff for the entry_ in a little similar capacity to a diff suppress func.

Using this setfunc when looking at the old and new state for a set, Terraform would hash all of the entries based on their name alone, and would treat any entries with matching names as identical even if their `expiryDetail` was different. It doesn't match entries to each other *and then* compare, it just matches entries.

We can safely remove this, and use the default- it'll do the right thing.

**Review by rileykarson — COMMENTED** [review]
Other than removing the set hash above, LGTM.

**Review by rileykarson — COMMENTED** [review]
Other than removing the set hash above, LGTM.

File: mmv1/products/cloudidentity/terraform.yaml (hunk 1 above)
Comment (tmshn) [reply mmv1/products/cloudidentity/terraform.yaml]: > Using this setfunc (...snip...) any entries with matching names as identical even if their `expiryDetail` was different.

Oh I see, that rings a bell. Removed dca9ddb3059ffe60b9099d9ace3c52ad2c996c1b

**Review by rileykarson — APPROVED** [review]

**Pull request merged**